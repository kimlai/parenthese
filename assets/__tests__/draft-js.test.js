import { convertToRaw, convertFromHTML, ContentState } from "draft-js";

test("convertToRaw matches the format saved in our database", () => {
  const html = "héllo <b>world</b>"; // using é to have a unicode character

  const blocksFromHTML = convertFromHTML(html);
  const state = ContentState.createFromBlockArray(
    blocksFromHTML.contentBlocks,
    blocksFromHTML.entityMap
  );

  const rawContent = convertToRaw(state);

  expect(rawContent).toMatchSnapshot({
    blocks: [
      {
        key: expect.any(String)
      }
    ]
  });
});
