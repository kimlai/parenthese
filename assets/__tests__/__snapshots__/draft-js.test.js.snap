// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToRaw matches the format saved in our database 1`] = `
Object {
  "blocks": Array [
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [
        Object {
          "length": 5,
          "offset": 7,
          "style": "BOLD",
        },
      ],
      "key": Any<String>,
      "text": "heÃÅllo world",
      "type": "unstyled",
    },
  ],
  "entityMap": Object {},
}
`;
